### PR TITLE
fix eeglab export rawarray filenames

### PR DESCRIPTION
**Pull request by 0reza**
#### Reference issue
No filenames Attribute in mne.io.RawArray #10921 

#### What does this implement/fix?
Add a try-except structure to catch the AttributeError and do nothing.

#### Additional information

### CODE REVIEW

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -111,6 +111,8 @@ Enhancements
 
 Bugs
 ~~~~
+- Fix bug in :func: `mne.export._eeglab._export_raw` to ignore empty filenames attribute in :class: `mne.io.RawArray` (:gh: `10927` by :newcontrib: `Reza Nasri`)
```
Comment (cbrnr) [inline doc/changes/latest.inc:114]:
```suggestion
- Fix bug in :func:`mne.export._eeglab._export_raw` to ignore empty filenames attribute in :class:`mne.io.RawArray` (:gh:`10927` by :newcontrib:`Reza Nasri`)
```

File: mne/export/_eeglab.py [hunk 2]
```diff
@@ -17,8 +17,11 @@ def _export_raw(fname, raw):
 
     # remove extra epoc and STI channels
     drop_chs = ['epoc']
-    if not (raw.filenames[0].endswith('.fif')):
-        drop_chs.append('STI 014')
+    try:
+        if not (raw.filenames[0].endswith('.fif')):
+            drop_chs.append('STI 014')
+    except AttributeError:  # mne.io.RawArray has no filenames attribute
+        pass
```
Comment (drammock) [inline mne/export/_eeglab.py:24]: I think I'd be more comfortable with:

```suggestion
    # RawArray doesn't have `filenames` attribute
    if hasattr(raw, 'filenames') and not raw.filenames[0].endswith('.fif'):
        drop_chs.append('STI 014')
```

the `except Error: pass` pattern always makes me nervous that silent bugs might slip through

Comment (0reza) [reply mne/export/_eeglab.py]: In fact, there is a misunderstanding by my comment.
RawArray has filenames attribute but RawArray.filenames[0]==None.

File: doc/changes/names.inc [hunk 3]
```diff
@@ -459,3 +459,5 @@
 .. _Frederik Weber: https://github.com/Frederik-D-Weber
 
 .. _Mingjian He: https://github.com/mh105
+
+.. _Reza Nasri: https://github.com/0reza
```
Comment (agramfort) [inline doc/changes/names.inc:463]:
```suggestion
.. _Reza Nasri: https://github.com/0reza

```

**Review by agramfort — APPROVED** [review]
just a nitpick

🙏 @0reza

**Review by cbrnr — APPROVED** [review]

**Pull request merged**